be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -255,13 +279,35 @@ Status BetaRowsetWriter::_rename_compacted_segment_plain(uint64_t seg_id) {
         _segid_statistics_map.emplace(_num_segcompacted, org);
         _clear_statistics_for_deleting_segments_unsafe(seg_id, seg_id);
     }
-    ++_num_segcompacted;
     ret = rename(src_seg_path.c_str(), dst_seg_path.c_str());
     if (ret) {
         LOG(WARNING) << "failed to rename " << src_seg_path << " to " << dst_seg_path
                      << ". ret:" << ret << " errno:" << errno;
         return Status::Error<ROWSET_RENAME_FILE_FAILED>();
     }
+    // rename remaining inverted index files
```

Comment:
the following code is similar to the code beginning with line 227 except for the path logic. Can we make a common function here to eliminate duplication?

Reply:
done